app/code/community/Sendsmaily/Sync/Helper/Data.php

```diff
@@ -41,98 +41,191 @@ public function getSyncData($collection=null) {
+    }
 
-        // Get customer lastname.
-        if (in_array('lastname', $fields)) {
-          $extra['lastname'] = $customer->getLastname();
-        }
+    // Get customer name prefix.
+    if (in_array('prefix', $fields)) {
+      $extra['prefix'] = $customer->getPrefix() ? $customer->getPrefix() : '';
+    }
 
-        // Get customer gender.
-        if (in_array('gender', $fields)) {
-          $attr = $customer->getAttribute('gender');
-          $extra['gender'] = $attr->getFrontend()
-            ->getOption($customer->getGender());
-        }
+    // Get customer group.
+    if (in_array('customer_group', $fields)) {
+      $code = Mage::getSingleton('customer/group')
+        ->load($customer->getGroupId())
+        ->getCode();
+      $extra['customer_group'] = $code;
+    }
 
-        // Get customer birthdate.
-        if (in_array('birthday', $fields)) {
-          $dob = $customer->getDob();
-          if (!empty($dob)) {
-            $extra['birthday'] = Mage::getModel('core/date')
-              ->date('Y-m-d', strtotime($dob));
-          }
-        }
+    // Get customer firstname.
+    if (in_array('firstname', $fields)) {
+      $extra['firstname'] = $customer->getFirstname();
+    }
 
-        // Clean up.
-        unset($customer);
+    // Get customer lastname.
+    if (in_array('lastname', $fields)) {
+      $extra['lastname'] = $customer->getLastname();
+    }
+
+    // Get customer gender.
+    if (in_array('gender', $fields)) {
+      $attr = $customer->getAttribute('gender');
+      $gender = $attr->getFrontend()->getOption($customer->getGender());
+      $extra['gender'] = $gender ? $gender : '';
+    }
+
+    // Get customer birthdate.
+    if (in_array('birthday', $fields)) {
+      $dob = $customer->getDob();
+      if (!empty($dob)) {
+        $extra['birthday'] = Mage::getSingleton('core/date')->date('Y-m-d', strtotime($dob));
       }
+    }
+
+    // Clean up.
+    unset($customer);
+
+    return $extra;
+  }
 
-      $data[] = array_merge($extra, array(
-        'email' => $item->hasSubscriberEmail() ? $item->getSubscriberEmail() : $item->getEmail(),
-        'subscription_status' => $item->getSubscriberStatus(),
-      ));
+  /**
+   * Get unsubscriber emails from Smaily.
+   *
+   * @param int $limit Limit batch size.
+   * @return array Unsubscriber emails list.
+   */
+  public function getUnsubscribersEmails($limit)
+  {
+    $curl = Mage::getModel('sync/curl');
+    $unsubscribersEmails = array();
+    $data = array(
+        'list' => 2,
+        'limit' => $limit,
+    );
+
+    while (true) {
+        $data['offset'] = $offset;
+        $unsubscribers = $curl->callApi('contact', $data);
+        if (!$unsubscribers) {
+            break;
+        }
+
+        foreach ($unsubscribers as $unsubscriber) {
+            $unsubscribersEmails[] = $unsubscriber['email'];
+        }
+
+        // Smaily api call offset is considered as page number, not sql offset!
```

Comment:
`api` should be `API` and `sql` should be `SQL`.

Reply:
Fixed in 608fbd51c1b00c0b7b2dfcd11dda1b068a5c5e16